### PR TITLE
AzureMonitor: Request and concat subsequent resource pages

### LINKED ISSUE #36586: AzureMonitor: New Logs resource picker displays only first 1000 resources - v8.0.x
My non-prod subscriptions (38) contain a few dozen Log Analytics workspaces.  In 7.x, the workspace selector was adequate.  I was able to see all subscriptions and all workspaces.  It was not necessary to see any other (non-Log Analytics) resources.

image

In the latest 8.x versions, the resource picker displays ALL resources that the identity has access to.  In non-prod, this is 1953 items.  In the new resource picker, only the first 1000 items are displayed.  This leaves out about half of the resources and makes them unselectable.  The situation is worse in Production where my total resource count is over 7500.  I am rolling back to 7.5.8 until this bug is resolved.

image

### PR DESCRIPTION
**Pull request by andresmgot**
**What this PR does / why we need it**:

When there is more than 1000 resources, only the first set of resources is fetched. We can detect if there are still resources to fetch and query additional pages. In this example, I changed the limit to 10, you can see that there are several requests, each one fetching a page:

https://user-images.githubusercontent.com/4025665/126353550-57ec11aa-86e3-41a1-819f-a88ac530d94b.mp4

Also, since this may take a while if there are many resources, I have added a loading message to the ResourcePicker component.

Note that this is not the definitive solution, just a workaround to address #36586 before we implement proper pagination support.

**Which issue(s) this PR fixes**:

Fixes #36586

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by joshhunt** [comment]
That's what the `<LoadingPlaceholder />` looks like? 😬

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts [hunk 1]
```diff
@@ -54,9 +55,24 @@ export default class ResourcePickerData extends DataSourceWithBackend<AzureMonit
...
+    while (resourceResponse.$skipToken) {
+      // The response include several pages
+      resourceResponse = await this.makeResourceGraphRequest<RawAzureResourceGroupItem[]>(query, 1, {
+        $skipToken: resourceResponse.$skipToken,
+      });
+      if (!resourceResponse.data.length) {
+        throw new Error('unable to fetch resource details');
+      }
+      resources = resources.concat(resourceResponse.data);
+    }
```
Comment (joshhunt) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/resourcePicker/resourcePickerData.ts:73]: maybe we could make this a single loop? something like

```typescript
let fetchMore = true;

while(fetchMore) {
  const response = ...;
  fetchMore = response.$skipToken

```
or something else fancier? Thoughts?

**Review by joshhunt — COMMENTED** [review]
Nice! Let's chat about the UI to see what we can do about that loading state :)

**Review by joshhunt — APPROVED** [review]

**Comment by andresmgot** [comment]
> The backport to `v8.0.x` failed:

I'll skip this since it will be released in 8.1 anyway.

**Pull request merged**